mteb/abstasks/TaskMetadata.py

```diff
@@ -180,3 +187,58 @@ def _check_dataset_revision_is_specified(cls, dataset):
                 "It is encourage to specify a dataset revision for reproducability"
             )
         return dataset
+
+    @field_validator("eval_langs")
+    def _check_eval_langs(cls, eval_langs):
+        """
+        This method checks that the eval_langs are specified as a list of languages.
+        """
+        if isinstance(eval_langs, list):
+            for code in eval_langs:
+                cls.check_language_code(code)
+        elif isinstance(eval_langs, dict):
+            for langs in eval_langs.values():
+                for code in langs:
+                    cls.check_language_code(code)
+        return eval_langs
+
+    @staticmethod
+    def check_language_code(code):
+        """
+        This method checks that the language code (e.g. "eng-Latn") is valid.
+        """
+        lang, script = code.split("-")
+        if lang not in ISOCODE639_3_TO_LANGUAGE:
+            raise ValueError(f"Invalid language code: {lang}")
+        if script not in ISOCODE15924_TO_SCRIPT:
+            raise ValueError(f"Invalid script code: {script}")
```

Comment:
Excellent check! Might be worth it to point the user towards the valid values? E.g. point them to the file containing all valid scripts/langs.